enhancements/sig-policy/54-core-policy-module/README.md

```diff
@@ -0,0 +1,227 @@
+1. Common methods: behavior like selecting namespaces and emitting compliance
+events for the policy framework will have a common implementation. This will
+reduce bugs from inconsistency, and allow policy controllers to focus on their
+specific behavior, not the details of interacting with the framework.
+1. Automated tests to ensure the features/methods listed above work properly
+when used by custom controllers.
+1. Custom Controller tutorial: we will create and maintain documentation showing
+how to create a new custom policy controller, and use the main features of this
+module.
+
+### Non-Goals
+
+1. Custom Controller Template: we will not maintain a template/sdk like the 
+operator-sdk for new policy controllers.
+
+## Proposal
+
+Create a new repository in open-cluster-mangement-io to provide implementations
+of common APIs and functions for controllers to work with the policy framework,
+and describe the behavior of the policy framework, similar to some of the docs
+in https://github.com/stolostron/governance-policy-framework.
+
+### User Stories
+
+#### Story 1
+
+As a developer of a policy controller, I can ensure consistent behavior with
+other policies by incorporating the common API types. For example, if the
+behavior of wildcards in the common `spec.namespaceSelector` field changes, I
+should not have to adjust my controller's code to be consistent, other than
+updating the version of the common library.
+
+#### Story 2
+
+As a creator of a new policy controller, I can focus on my controller's 
+specialized detection of violations, and rely on common implementations of
+functions in order to interact with the policy framework.
+
+#### Story 3
+
+As a developer of multiple policy controllers, I can fix a bug or add a feature
+to common policy behavior in one place, and not duplicate it in each controller.
+
+### Implementation Details/Notes/Constraints 
+
+Although kubebuilder will automatically pull API validation from embedded
+structs (like when types embed `metav1.TypeMeta`), it will not automatically
+pull RBAC permissions on methods into the project. So RBAC requirements on
+common functions (eg `get namespaces` for selecting namespaces) will need to
+be well-documented.
+
+This module can include features that aren't strictly required or currently used
+by the policy framework. For example, storing additional structured information
+about k8s objects that the policy looks at in a `status.relatedObjects` field.
+Experimental features can be added, but should be marked as such by the
+versioning of the repository.
+
+### Risks and Mitigation
+
+(unknown)
+
+## Design Details
+
+Here is a part of the common struct and validation from a
+[prototype](https://github.com/JustinKuli/policycore-test/blob/main/api/v1/policycore_types.go)
+of this proposal:
+
+```golang
+type PolicyCoreSpec struct {
+	Severity          Severity          `json:"severity,omitempty"`
```

Comment:
It is necessary to doc the behavior of each field here. And I am not sure how to ensure two different customized policy controllers behave the same for these fields.

Reply:
I agree that would be documented in the actual module. Should I put all those details in the proposal? I agree they could use more explanation, but I think we would start from what is currently documented in the configuration policy: https://github.com/open-cluster-management-io/config-policy-controller/blob/main/api/v1/configurationpolicy_types.go

For consistent behavior, for some things I think we can only document. For example, the severity field is currently just informational in the controllers - other things like alerting systems could use it but it shouldn't change the behavior of the policy. But NamespaceSelector would have a method that all the controllers can use.

Reply:
I think if there are two controllers that embed this spec in their own spec. how they behave on these fields is questionable. I wonder if we will have a common agent that consumes any customized policy and consumes field e.g severity. Maybe some description here will be helpful.